### PR TITLE
DashboardScene: Support auto migration for angular panels

### PR DESCRIPTION
**Pull request by torkelo**
Uses new custom handler added in https://github.com/grafana/scenes/pull/396 

This replicates the logic found in PanelModel pluginLoaded that handles auto migration from old angular panels to react panels. 
https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/state/PanelModel.ts#L440 

I know this is a bit messy as it's based on passing around this panel model-like object that is mutated (options & fieldConfig are set). But since this is how the migration handlers currently work it simplifies things to keep it that way, and this should be "temporary" :)

### CODE REVIEW

File: public/app/features/dashboard-scene/serialization/angularMigration.test.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+      const reactPlugin = getPanelPlugin({ id: 'timeseries' }).setPanelChangeHandler(onPanelTypeChanged as any);
+
+      const oldModel = new PanelModel({
+        autoMigrateFrom: 'graph',
+        oldOptionProp: 'old name',
+        type: 'timeseries',
+      });
+
+      const mutedModel = {
```
Comment (dprokop) [inline public/app/features/dashboard-scene/serialization/angularMigration.test.ts:22]: heh, I guess you meant: 
```suggestion
      const mutatedModel = {
```

File: public/app/features/dashboard-scene/serialization/angularMigration.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      const oldOptions = oldModel.getOptionsToRemember();
+      const prevPluginId = oldModel.autoMigrateFrom;
+
+      if (plugin.onPanelTypeChanged) {
+        const prevOptions = wasAngular ? { angular: oldOptions } : oldOptions.options;
+        Object.assign(panel.options, plugin.onPanelTypeChanged(panel, prevPluginId, prevOptions, panel.fieldConfig));
+      }
+    }
+
+    return {};
```
Comment (dprokop) [inline public/app/features/dashboard-scene/serialization/angularMigration.ts:22]: this probably deserves a void return? 
```suggestion
    return;
```

**Review by dprokop — APPROVED** [review]
LGTM - just cosider my comments in https://github.com/grafana/scenes/pull/396#discussion_r1351950794

File: public/app/features/dashboard-scene/serialization/angularMigration.ts (hunk 2 above)
Comment (torkelo) [reply public/app/features/dashboard-scene/serialization/angularMigration.ts:22]: yes :)

**Pull request merged**